src/main/java/seedu/address/logic/commands/UpdateCommand.java

```diff
@@ -54,60 +55,68 @@ public class EditCommand extends Command {
             + PREFIX_PHONE + "91234567 "
             + PREFIX_EMAIL + "johndoe@example.com";
 
-    public static final String MESSAGE_EDIT_PERSON_SUCCESS = "Edited Person: %1$s";
-    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_UPDATE_PERSON_SUCCESS = "Updated Person: %1$s";
+    public static final String MESSAGE_NOT_UPDATED = "At least one field to update must be provided.";
     public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book.";
 
-    private final Index index;
-    private final EditPersonDescriptor editPersonDescriptor;
+    private final Name name;
+    private final Name lowerCaseName;
+    private final UpdatePersonDescriptor updatePersonDescriptor;
 
     /**
-     * @param index of the person in the filtered person list to edit
-     * @param editPersonDescriptor details to edit the person with
+     * @param name of the person in the filtered person list to update
+     * @param updatePersonDescriptor details to update the person with
      */
-    public EditCommand(Index index, EditPersonDescriptor editPersonDescriptor) {
-        requireNonNull(index);
-        requireNonNull(editPersonDescriptor);
+    public UpdateCommand(Name name, UpdatePersonDescriptor updatePersonDescriptor) {
+        requireNonNull(name);
+        requireNonNull(updatePersonDescriptor);
 
-        this.index = index;
-        this.editPersonDescriptor = new EditPersonDescriptor(editPersonDescriptor);
+        this.name = name;
+        this.lowerCaseName = new Name(name.fullName.toLowerCase());
+        this.updatePersonDescriptor = new UpdatePersonDescriptor(updatePersonDescriptor);
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Person> lastShownList = model.getFilteredPersonList();
+        List<Person> lastShownList = model.getAddressBook().getPersonList();
 
-        if (index.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        List<Person> sameNamePeople = lastShownList.stream()
+                .filter(person -> person.getLowerCaseName().equals(lowerCaseName))
+                .collect(Collectors.toList());
+
+        if (sameNamePeople.size() != 1) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_NAME);
```

Comment:
I suggest splitting the condition to more than one you can split the error messages to be for when the sameNamePeople is empty (so the name given matches no one, and it is invalid) and when the sameNamePeople is more than one (there are duplicates in the list)